airflow/models/mappedoperator.py

```diff
@@ -687,7 +687,7 @@ def render_template_fields(
         unmapped_task = self.unmap(mapped_kwargs)
         context_update_for_unmapped(context, unmapped_task)
 
-        self._do_render_template_fields(
+        unmapped_task._do_render_template_fields(
```

Comment:
A comment here containing the explaination you gave in the PR description would be a good idea.